### PR TITLE
[0.15] Update release notes for 0.15.1

### PR DESCRIPTION
**Pull request by meshcollider**
Adds info for the more notable changes since 0.15.0.1 to the release notes.

~Still needs the list of commits added too.~ (added in https://github.com/bitcoin/bitcoin/commit/2ce9e586d5b11e1f61af4db27b7326687d5e5124)

### CODE REVIEW

**Comment by fanquake** [comment]
Needs a rebase after 2ce9e586d5b11e1f61af4db27b7326687d5e5124

**Comment by laanwj** [comment]
Thanks, I hadn't noticed this, though I think the only work we duplicated is the credits list.

File: doc/release-notes.md [hunk 1]
```diff
@@ -84,15 +105,80 @@ implied blockmaxweight, instead of limiting block size directly. Any miners who
...
+
 - The "currentblocksize" value in getmininginfo has been removed.
 
+- `dumpwallet` no longer allows overwriting files. This is a security measure
+  as well as prevents dangerous user mistakes.
+
+- `backupwallet` will now fail when attempting to backup to source file, rather than
+  destroying the wallet.
+
+- `listsinceblock` will now throw an error if an unknown `blockhash` argument
```
Comment (TheBlueMatt) [inline doc/release-notes.md]: May want to clarify that its still allowed to be an empty string, which will still have the old behavior.

**Comment by MeshCollider** [comment]
Rebased + added the note suggested by @TheBlueMatt 

I've cancelled most of the travis jobs because no point running the build+tests every time I push a fixup commit for a doc-only change :)

**Pull request merged**